File: src/main/java/com/example/demo/poc/inspection/dto/PocUserInspectionResponse.java
package com.example.demo.poc.inspection.dto;

import com.fasterxml.jackson.annotation.JsonInclude;

import java.time.Instant;

public record PocUserInspectionResponse(
        String id,
        String username,
        @JsonInclude(JsonInclude.Include.ALWAYS)
        String nickname,
        String passwordHash,
        Instant createdAt,
        Instant updatedAt,
        @JsonInclude(JsonInclude.Include.ALWAYS)
        Instant deletedAt,
        @JsonInclude(JsonInclude.Include.ALWAYS)
        String activeUsernameKey,
        @JsonInclude(JsonInclude.Include.ALWAYS)
        String activeNicknameKey
) {
}

File: src/main/java/com/example/demo/poc/inspection/dto/PocUserInspectionResponse.kt
package com.example.demo.poc.inspection.dto

import com.fasterxml.jackson.annotation.JsonInclude
import java.time.Instant

@JvmRecord
data class PocUserInspectionResponse(
    val id: String,
    val username: String,
    @get:JsonInclude(JsonInclude.Include.ALWAYS)
    val nickname: String?,
    val passwordHash: String,
    val createdAt: Instant,
    val updatedAt: Instant,
    @get:JsonInclude(JsonInclude.Include.ALWAYS)
    val deletedAt: Instant?,
    @get:JsonInclude(JsonInclude.Include.ALWAYS)
    val activeUsernameKey: String?,
    @get:JsonInclude(JsonInclude.Include.ALWAYS)
    val activeNicknameKey: String?
)
